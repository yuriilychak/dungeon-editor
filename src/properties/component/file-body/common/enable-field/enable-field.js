import React, {memo, useState, useCallback, useEffect} from "react";

import {CheckBox} from "../check-box";

const EnableField = ({
                         labelEnabled,
                         id,
                         value,
                         onChange,
                         children
                     }) => {
    const [enabled, setEnabled] = useState(false);

    useEffect(() => setEnabled(value), [value]);

    const handleChange = useCallback(data => {
        setEnabled(data.value);
        onChange(data);
    }, [onChange]);

    return (
        <>
            <CheckBox
                id={id}
                label={labelEnabled}
                checked={enabled}
                onChange={handleChange}
            />
            {enabled && (
                <>
                    {children}
                </>
            )}
        </>
    )
};

export default memo(EnableField);
