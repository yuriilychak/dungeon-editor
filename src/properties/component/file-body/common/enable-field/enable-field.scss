@import "../../../../../sizes";

.properties-enable-field {
  display: flex;
  flex-direction: row;

  & > * {
    margin-right: $indent_l;
    &:last-child {
      margin-right: 0;
    }
  }
}

.properties-enable-field-toggle-disabled {
  pointer-events: none;
  opacity: 0.5;
}

.properties-enable-field-deselected {
  pointer-events: none;
  opacity: 0.5;
}
