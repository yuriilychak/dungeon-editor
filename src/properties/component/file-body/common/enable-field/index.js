export { default as EnabledField } from "./enable-field";
