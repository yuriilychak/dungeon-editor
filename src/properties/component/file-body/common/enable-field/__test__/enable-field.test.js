import { createMount } from "@material-ui/core/test-utils";
import React from "react";

import { EnableField } from "../enable-field";

jest.mock("../../../../../../common-ui", () => ({
    ToolButton: ({ onClick }) => (
        <button
            onClick={onClick}
        >
        Mock Colorpicker
        </button>
    )
}));

describe("GIVEN EnableField from property section", () => {
    const mockChange = jest.fn();
    const wrapper = createMount()(
        <EnableField
            id="test"
            value={true}
            label="test"
            onChange={mockChange}
        />
    );

    it("WHEN render component with default props THEN component should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("WHEN dispatch onChange THAN result should equal expected object", () => {
        wrapper.find("button").simulate("click");
        expect(mockChange).toHaveBeenCalledWith(false);
    });
});
