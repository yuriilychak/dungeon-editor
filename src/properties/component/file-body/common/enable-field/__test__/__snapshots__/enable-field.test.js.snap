// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN EnableField from property section WHEN render component with default props THEN component should match snapshot 1`] = `"<button>Mock Colorpicker</button><div class=\\"properties-enable-field\\"></div>"`;
